### ISSUE TITLE
closed-loop MPC

### ISSUE THREAD

**Issue by Saeed-Mansouri** [jrl-umi3218/lipm_walking_controller#40]
Hi @stephane-caron,

You have implemented an open-loop MPC to generate the desired com trajectory and a DCM-feedback controller (PID) to make the ZMP command for tracking the desired DCM. Why didn't you implement a closed-loop MPC to directly generate the ZMP command? I'd like to know if there are any issues or challenges to having a closed-loop MPC (instead of having an open-loop MPC + DCM feedback controller).

**Comment by stephane-caron** [comment]
Yes, both approaches have their pros and cons. I've drafted definitions and pointers to the works I know in the following post: [Open loop and closed loop model predictive control](https://scaron.info/robot-locomotion/open-closed-loop-model-predictive-control.html). Don't hesitate to join the discussion if you find some other works or have some robot experience to share.
